src/server/game/AI/SmartScripts/SmartAI.h

```diff
@@ -123,6 +123,15 @@ class TC_GAME_API SmartAI : public CreatureAI
         // Called when spell hits a target
         void SpellHitTarget(WorldObject* target, SpellInfo const* spellInfo) override;
 
+        // Called when a spell finishes
+        void OnSpellCast(SpellInfo const* spellInfo) override;
```

Comment:
Then mby OnSpellFinished for naming consistency?

Reply:
https://wow.tools/dbc/?dbc=enumeratedstring&build=2.5.1.39170#page=1&colFilter[1]=spellcast&colFilter[4]=598